client/allochealth/tracker.go

```diff
@@ -262,7 +308,7 @@ func (t *Tracker) watchTaskEvents() {
 		}
 
 		// Store the task states
-		t.l.Lock()
+		t.lock.Lock()
 		for task, state := range alloc.TaskStates {
 			//TODO(schmichael) for now skip unknown tasks as
```

Comment:
```
			//TODO(schmichael)
```

The scariest words I can see in a PR. I don't even think this comment is accurate. It seems to be copied and pasted from another place, but here we're iterating over `alloc.TaskStates` and updating a map that was originally populated from `alloc.TaskStates` ... I really don't see how group services could factor into this?

If you have 30 seconds of time to give this comment a think and remove it if you think it's nonsensical as well, I'd appreciate you cleaning up past schmichael's messes.

I don't see how `taskHealth[task]` could ever be `!ok`, but we don't have to worry about changing the actual code too.

Reply:
Heh yeah I was wondering about this :sweat_smile: